app/components/TransactionHistory/TransactionHistoryPanel/TransactionHistoryPanel.jsx

```diff
@@ -24,11 +30,39 @@ export default class TransactionHistory extends React.Component<Props> {
         <Transactions
           className={styles.transactions}
           transactions={transactions}
+          pendingTransactions={pendingTransactions || []}
         />
       </Panel>
     )
   }
 
+  async pruneReturnedTransactionsFromStorage() {
+    const {
+      transactions,
+      pendingTransactions,
+      address,
+      net,
+      getPendingTransactionInfo,
+    } = this.props
+    if (
+      transactions &&
+      transactions.length &&
+      pendingTransactions &&
+      pendingTransactions.length
+    ) {
+      const confirmed = transactions.map(tx => tx.txid)
+      // NOTE: removes the '0x' prepended to every txId
+      const pending = pendingTransactions.map(tx => tx.txid.substring(2))
+      const toBePurged = intersection(confirmed, pending)
```

Comment:
nit: if `pendingTransactions.txId` would have been normalized beforehand (is there any reason not to?) you could do this in one line:
```
const toBePurged = intersectionBy(transactions, pendingTransactions, 'txId')
```

Reply:
I like this idea and implemented it however it caused me to change the filter logic here:

```
export const pruneConfirmedOrStaleTransaction = async (
  address: string,
  txId: string,
) => {
  const storage = await getPendingTransactions()
  if (Array.isArray(storage[address])) {
    storage[address] = storage[address].filter(
      // use includes here to be indifferent to 0x prefix
      transaction => !transaction.hash.includes(txId),
    )
  }
  await setPendingTransactions(storage)
}
```